test/ValidateID.cpp

```diff
@@ -378,14 +383,93 @@ TEST_F(ValidateID, OpTypeArrayElementTypeBad) {
 %3 = OpTypeArray %2 %2)";
   CHECK(spirv, SPV_ERROR_INVALID_ID);
 }
-TEST_F(ValidateID, OpTypeArrayLengthBad) {
+
+// Signed or unsigned.
+enum Signed { kSigned, kUnsigned };
+
+// Creates an assembly snippet declaring OpTypeArray with the given length.
+string MakeArrayLength(const string& len, Signed isSigned, int width) {
+  ostringstream ss;
+  ss << " %t = OpTypeInt " << width << (isSigned == kSigned ? " 1" : " 0")
+     << " %l = OpConstant %t " << len << " %a = OpTypeArray %t %l";
+  return ss.str();
+}
+
+TEST_F(ValidateID, OpTypeArrayLength0) {
```

Comment:
This next run of tests are all invalid cases.
How about some positive test cases for lengths that are constant integers: signed case, and also signed/unsigned wider than 32 bits (e.g. 48 and 64)